landscape/package/tests/test_reporter.py

```diff
@@ -926,6 +933,51 @@ def got_result(result):
         result = self.reporter.detect_packages_changes()
         return result.addCallback(got_result)
 
+    def test_detect_packages_changes_with_autoremovable(self):
+        message_store = self.broker_service.message_store
+        message_store.set_accepted_types(["packages"])
+
+        self.store.set_hash_ids({HASH1: 1, HASH2: 2, HASH3: 3})
+        self.store.add_available([1, 2, 3])
+        self.store.add_installed([1])
+        self.set_pkg1_autoremovable()
+
+        result = self.successResultOf(self.reporter.detect_packages_changes())
+        self.assertTrue(result)
+
+        expected = [{"type": "packages", "autoremovable": [1]}]
+        self.assertMessages(message_store.get_pending_messages(), expected)
+        self.assertEqual([1], self.store.get_autoremovable())
+
+    def test_detect_packages_changes_with_not_autoremovable(self):
+        message_store = self.broker_service.message_store
+        message_store.set_accepted_types(["packages"])
+
+        self.store.set_hash_ids({HASH1: 1, HASH2: 2, HASH3: 3})
+        self.store.add_available([1, 2, 3])
+        self.store.add_autoremovable([1, 2])
```

Comment:
It may be worth adding a note here pointing out that add_installed() is never called, which is why not-autoremovable is what it is.

FWIW, I also find this approach to exercising not-autoremovable to be a little odd.  At what point would the store mark a package as auto-removable but not installed?